src/server/controllers/signUp.js

```diff
@@ -1,3 +1,33 @@
-async function signUp(req, res, next) {}
+import { nanoid } from 'nanoid';
+import { sendEmail } from '../handlers/sendEmail.js';
+import { User } from '../models/schema/users/userSchema.js';
+
+async function signUp(req, res) {
+  try {
+    const { email, password, name } = req.body;
+    if (await User.findOne({ email })) {
+      return res
+        .status(409)
+        .json({ message: 'User with this email already exist.' });
+    }
+    const user = new User({ email });
+    user.setPassword(password);
+    user.name = name;
+    // user.avatarURL = gravatar.url(email, { s: 250, protocol: "https" });
+    // question if we will use it
+    const verificationToken = nanoid();
+    user.verificationToken = verificationToken;
+    sendEmail({
+      to: user.email,
+      subject: 'SoYummy - confirm your email!',
+      text: 'Welcome in SoYummy! To verify your account please click this link:',
+      html: `<a href="http://localhost:3000/api/users/verify/${verificationToken}">Confirm account</a>`,
+    });
```

Comment:
Looks like we are using sendEmail in two files with same properties. It will be good idea to create helpful function and just reuse it in both cases passing thrue only `email` and `token`.
It will keep our code clean.